mantidproject/mantid
# Ed87 workspace creation speed improvement

**Description of work.**
This pr adds some quick and effective performance improvements to FilterEvents.  As suggested by @rosswhitfield 

- [x] pr against `ornl-next` 

Companion to PR ornl-next [![GitHub issue/pull request detail](https://img.shields.io/github/issues/detail/state/mantidproject/mantid/32998)](https://github.com/mantidproject/mantid/pull/32998) 



**To test:**

you can run the following script before and after the changes to see the performance gains.

```

# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

LoadEventNexus(Filename='/SNS/VULCAN/IPTS-26944/nexus/VULCAN_195855.nxs.h5', OutputWorkspace='VUL195855')
CreateGroupingWorkspace(InputWorkspace='VUL195855', GroupDetectorsBy='bank', OutputWorkspace='VULCAN_group')
AlignAndFocusPowder(InputWorkspace='VUL195855', OutputWorkspace='VUL195855', GroupingWorkspace='VULCAN_group',
                    Params=(.1,-0.004,2),
                    CompressWallClockTolerance=1) # seconds

GenerateEventsFilter(InputWorkspace='VUL195855', OutputWorkspace='100s_chop', InformationWorkspace='100s_chop_info', TimeInterval='100')
FilterEvents(InputWorkspace='VUL195855', SplitterWorkspace='100s_chop', OutputWorkspaceBaseName='output_100s',GroupWorkspaces=True)


```

#### Release Notes

Diffraction Release Notes 6.3.0v

Improvements
############
- FilterEvents  execution speed improved by 35% in some cases.


#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
